File: bindings/dotnet/src/Aster/Registry/HealthStatus.cs
namespace Aster.Registry;

/// <summary>
/// Health state of a registered endpoint (Aster-SPEC.md §11.6).
/// </summary>
/// <remarks>
/// Modeled as a constants class (not enum) because the wire format uses lowercase
/// string values that pass through the registry doc and FFI layer untouched.
/// </remarks>
public static class HealthStatus
{
    public const string Starting = "starting";
    public const string Ready = "ready";
    public const string Degraded = "degraded";
    public const string Draining = "draining";

    public static bool IsValid(string value) =>
        value is Starting or Ready or Degraded or Draining;

    public static string Validate(string value)
    {
        if (!IsValid(value))
            throw new ArgumentException($"Invalid HealthStatus: {value}", nameof(value));
        return value;
    }

    public static bool IsRoutable(string status) =>
        status is Ready or Degraded;
}

File: bindings/java/aster-runtime/src/main/java/site/aster/registry/HealthStatus.java
package site.aster.registry;

import java.util.Set;

/**
 * Health state of a registered endpoint (Aster-SPEC.md §11.6).
 *
 * <p>Modeled as a constants class (not enum) because the wire format uses lowercase string values
 * that pass through the registry doc and FFI layer untouched.
 */
public final class HealthStatus {

  public static final String STARTING = "starting";
  public static final String READY = "ready";
  public static final String DEGRADED = "degraded";
  public static final String DRAINING = "draining";

  private static final Set<String> VALID = Set.of(STARTING, READY, DEGRADED, DRAINING);

  private HealthStatus() {}

  public static String validate(String value) {
    if (!VALID.contains(value)) {
      throw new IllegalArgumentException("Invalid HealthStatus: " + value);
    }
    return value;
  }

  public static boolean isRoutable(String status) {
    return READY.equals(status) || DEGRADED.equals(status);
  }
}
